unittests/Packages/ifthen.py

```diff
@@ -0,0 +1,268 @@
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-true' in str(output)
+    text = '''\\ifthenelse{\\lengthtest{1cm=11mm}}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-false' in str(output)
+
+def test_ifthenelse_and(tmpdir, renderXHTML, make_document):
+    text = '''\\ifthenelse{3>2 \\and 5>4}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-true' in str(output)
+    text = '''\\ifthenelse{3>2 \\AND 5>4}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-true' in str(output)
+    text = '''\\ifthenelse{3>2 \\and 5>40}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-false' in str(output)
+    text = '''\\ifthenelse{3>20 \\AND 5>4}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-false' in str(output)
+
+def test_ifthenelse_or(tmpdir, renderXHTML, make_document):
+    text = '''\\ifthenelse{3>2 \\or 5>4}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-true' in str(output)
+    text = '''\\ifthenelse{3>20 \\OR 5>4}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-true' in str(output)
+    text = '''\\ifthenelse{3>20 \\or 5>40}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-false' in str(output)
+    text = '''\\ifthenelse{3>20 \\OR 5>40}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
+    assert 'ifthen-false' in str(output)
+
+def test_ifthenelse_not(tmpdir, renderXHTML, make_document):
+    text = '''\\ifthenelse{\\not 5>40}{ifthen-true}{ifthen-false}'''
+    doc = make_document(packages='ifthen', content=text)
+    out = doc.getElementsByTagName('document')[0]
+    output = renderXHTML(tmpdir, doc)
```

Comment:
Is there any reason to run a renderer here? Can't you simply `assert 'ifthen-true' in out.textContent`? Running the renderer will take time, create temporary files and read them. We already have many tests that are slow but we can't easily avoid them. Slow test suites create lazy programmers who push without running the test suite...

Reply:
I've removed the renderers in the unit tests. It runs *much* faster now, thank you.